LC 3279 Reporting date picker

jaylco opened the pull request:
- Create new date picker nunjucks macro
- Implement new macro in the existing reporting interface
- Improve error display/readability in existing reporting interface
- Implement new date picker macro as a 'custom' date option within the reporting MVP dashboard

SavvasCO reviewed (commented):
Seems like the "The start date must be before the end date" error shows up when I add an invalid date (for example "32/11") even if the start date is (technically) before the end date (start "1/11/2024", end "32/11/2024"):

![Screenshot 2024-12-13 at 14 49 48](https://github.com/user-attachments/assets/03e6ea5d-936f-40c8-b774-0ede26b75ae0)

Also, seems we decided to remove the values from the input fields in booking info / learning record reports, should we do the same here?

jaylco commented:
- Good spot! Fixed
- We removed the default dates for the CSV reports because they defaulted to today's date for both and thus weren't very helpful. The new MVP datepicker will 'remember' the dates via the session like it does for all of the selections the user has made. I think there's definitely a discussion to be had (based on feedback of course) around whether or not this behaviour is helpful
